src/nvim/getchar.c

```diff
@@ -2015,6 +2036,12 @@ static int vgetorpeek(int advance)
             else {
               int noremap;
 
+              // If this is a LANGMAP mapping, then we didn't record the keys
+              // at the start of the function and have to record them now.
+              if (keylen > typebuf.tb_maplen && (mp->m_mode & LANGMAP) != 0) {
+                gotchars(s, -1);
```

Comment:
No problem.

Just to note: for me LANGMAP is a very common case (~30% of keys in insert mode), but the map expansions are all one character long so it's still not a performance problem :-)